### ISSUE TITLE
Accessibility issue

### ISSUE THREAD

**Issue by emmarichardson** [Lesterhuis-Training-en-Consultancy/moodle-block_quickcourselist#1]
I am receiving an error on an accessibility scan that the form is missing a label - the error is pointing to this in the html.
<input autocomplete="off" name="efquicklistsearch" id="efquicklistsearch" class="form-control" value="">

**Comment by emmarichardson** [comment]
For some reason, the code I pasted above did not stick.  The label that is needed is to show that the form should not pull from autocomplete - so it is needing an aria label for the input box.
In addition, accessibility is complaining about lack of contrast with the form box.  The border is too light.

**Comment by emmarichardson** [comment]
Here is a screenshot with the contrast issue - I think the form labelling is not needed - as I read more that was in case it should have an autofill which it should not.
<img width="400" alt="Screenshot 2024-03-20 at 8 01 00 AM" src="https://github.com/Lesterhuis-Training-en-Consultancy/moodle-block_quickcourselist/assets/2079129/d3b0e593-9d08-4a5b-86b5-cceb40f1482a">

**Comment by gemguardian** [comment]
@luukverhoeven could you please put this on your calendar to check how much time is needed to fix it.

**Comment by luukverhoeven** [comment]
@vincentcornelis Can you check this? Thanks

**Comment by vincentcornelis** [comment]
Hi @emmarichardson 

What Moodle version are you receiving this accessibility issue for?
Are you using theme Boost, or another theme?

Correct me if i'm wrong, but this issue would be regarding the following element?: 
```html
<input ... name="efquicklistsearch">...</input>
``` 
Default display of this element has the text-input as white on white with a dark-gray border, which when focussed gives the additional blue outline, which should provide a contrast of 3.02 for the gray, (which, i realise, would still not reach the aforementioned 3.1 contrast. However, we don't apply additional styling, this is the case for all default Moodle form elements), and a contrast of 3.36 for the blue.

![image](https://github.com/Lesterhuis-Training-en-Consultancy/moodle-block_quickcourselist/assets/86881597/4d49f171-4572-4fcc-8b18-0454e01a61c9)

![image](https://github.com/Lesterhuis-Training-en-Consultancy/moodle-block_quickcourselist/assets/86881597/8d5fd72a-338f-4e9f-99e2-963542287dee)


### Moodle 4.2:
![image](https://github.com/Lesterhuis-Training-en-Consultancy/moodle-block_quickcourselist/assets/86881597/dc2571f4-9015-4530-82e4-67f0d1d56e44)
![image](https://github.com/Lesterhuis-Training-en-Consultancy/moodle-block_quickcourselist/assets/86881597/979ba038-7faa-44ff-9965-baf2df0ad517)

### Moodle 3.11:
![image](https://github.com/Lesterhuis-Training-en-Consultancy/moodle-block_quickcourselist/assets/86881597/757abcdf-7601-47b9-9dac-d3caf4221c27)
![image](https://github.com/Lesterhuis-Training-en-Consultancy/moodle-block_quickcourselist/assets/86881597/bf0ef0a6-e11d-41bc-86e4-85fd7a8f04eb)

I am however unable to discover a place where the contrast is the 1.49:1 you mentioned.

Like to hear back from you!

**Comment by luukverhoeven** [comment]
@gemguardian @emmarichardson Do you have an update on this?

**Comment by gemguardian** [comment]
@luukverhoeven as it seems we cannot replicate this issue, or received other remarks on it. I tend to close this issue as 'won't fix'. @emmarichardson if you have time to reply we dan take a nother look at it.

**Comment by emmarichardson** [comment]
Sorry for lack of response.  I ended up using css to get around the contrast issue.  Using Foundation as theme.  I still receive the complaint about the lack of label for aria but have yet to determine if that is an actual issue because it is expecting a label for autofill which is not appropriate.

**Comment by gemguardian** [comment]
@emmarichardson thank you for your response, since we use the stying of Moodle Boost form, and don't 'override' the styling, this sounds as something you could make a Moodle tracker of so that moodle pty ltd is aware of this accessibility issue. 

as for the lack of label aria, I checked the wcag 1.2 guidelines and I do think we can assist in this - but we only deploy it for the latest master/main branche. 

@luukverhoeven I would suggest the Aria Seach because it is a search form https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/searchbox_role or - following the fact we use form - we could use the aria form : https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/form_role

Could you please let me know what you think and the time invested to fix for the main /master branche?

**Comment by luukverhoeven** [comment]
@gemguardian We estimate that it will take 1 or 2 hours to develop and test this.

**Comment by gemguardian** [comment]
@luukverhoeven This is approved but for the latest main branche version, we don't backport this issue.

**Comment by luukverhoeven** [comment]
@gemguardian Can you check if the following branch (https://github.com/Lesterhuis-Training-en-Consultancy/moodle-block_quickcourselist/tree/86byan3tr-accessibility-fix)

**Comment by gemguardian** [comment]
@luukverhoeven I can confirm that we see a aria-label "Search for courses" in M4.4 

![image](https://github.com/Lesterhuis-Training-en-Consultancy/moodle-block_quickcourselist/assets/16559999/86e8a7d4-e078-45d4-ae5e-138c2fc55b2c)

**Comment by gemguardian** [comment]
@luukverhoeven could yuo please this to the master and release it on Moodle.org  - Could you please check Moodle.org branches as well, I doubt that this is the same line as we use Github. 

- Gemma

**Comment by luukverhoeven** [comment]
@gemguardian Merged and published on Moodle.org
